app/controllers/coronavirus/sub_sections_controller.rb

```diff
@@ -9,11 +9,22 @@ def new
 
     def create
       @sub_section = page.sub_sections.new(sub_section_params)
-      if @sub_section.save && draft_updater.send
-        redirect_to coronavirus_page_path(page.slug), notice: "Sub-section was successfully created."
-      else
-        @sub_section.errors.add :base, draft_updater.errors.to_sentence
+      errors = nil
+
+      SubSection.transaction do
+        if @sub_section.save
+          unless draft_updater.send
+            errors = draft_updater.errors.to_sentence
+            raise ActiveRecord::Rollback
+          end
+        end
+      end
+
+      if !@sub_section.created_at || errors
+        @sub_section.errors.add :base, errors
```

Comment:
Can the error be rendered using an `flash["alert"]` instead? 

e.g.
```
if draft_updater.errors.any?
  flash["alert"] = draft_updater.errors.to_sentence
  render :new
end
```

Reply:
Sorted after a talk in person 🎉